conans/client/build/cmake_flags.py

```diff
@@ -152,12 +152,11 @@ def _cmake_cross_build_defines(self):
             ret["CMAKE_SYSTEM_NAME"] = cmake_system_name
         else:  # detect if we are cross building and the system name and version
             if cross_building(self._conanfile.settings):  # We are cross building
-                if os_ != os_build:
-                    if os_:  # the_os is the host (regular setting)
-                        ret["CMAKE_SYSTEM_NAME"] = ("Darwin" if os_ in ["iOS", "tvOS", "watchOS"]
-                                                    else os_)
-                    else:
-                        ret["CMAKE_SYSTEM_NAME"] = "Generic"
+                if os_:  # the_os is the host (regular setting)
+                    ret["CMAKE_SYSTEM_NAME"] = ("Darwin" if os_ in ["iOS", "tvOS", "watchOS"]
+                                                else os_)
+                else:
+                    ret["CMAKE_SYSTEM_NAME"] = "Generic"
```

Comment:
`cross_building` also returns True when x86_64 => x86 and I'm not sure we should adjust `CMAKE_SYSTEM_NAME` in that case.

Reply:
Reading this: https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html probably you are right. But I'm almost sure this produce some other issue, I'm trying to recall and asking people.

Reply:
browsing the history I'd say that this `if` should have been removed with https://github.com/conan-io/conan/commit/20eb12fe73c6ac2b9b2ab06a3fab684e83ae58a1#diff-029e992e62f056e56357fa0ba4cf4111R160, but it is true that this change can have some unexpected consequences on x86_64->x86 builds

Reply:
Yeah, my commit message `Not adjust system name when the os is the same but the arch is not the same` is disturbing. I did it for some reason... but it is a shame I didn't referece any issue.

Reply:
One of the thing which breaks with when cross compiling from x86_64->x86 on same os is [try_run](https://cmake.org/cmake/help/latest/command/try_run.html#behavior-when-cross-compiling). The current PR should handle it correctly